src/Umbraco.Web.BackOffice/Trees/ContentTreeControllerBase.cs

```diff
@@ -429,6 +429,24 @@ protected sealed override ActionResult<TreeNodeCollection> GetTreeNodes(string i
             return nodes ?? new TreeNodeCollection();
         }
 
+        // if we're rendering the root and people don't have access to the root still grant them access to the recycle bin
+        else if (id == Constants.System.RootString && !UserStartNodes.Contains(Constants.System.Root))
+        {
+            var nodes = new TreeNodeCollection();
+
+            nodes?.Add(CreateTreeNode(
+                    RecycleBinId.ToInvariantString(),
+                    id,
+                    queryStrings,
+                    LocalizedTextService.Localize("general", "recycleBin"),
+                    "icon-trash",
+                    RecycleBinSmells,
+                    queryStrings.GetRequiredValue<string>("application") + TreeAlias.EnsureStartsWith('/') +
+                    "/recyclebin"));
+
+            return nodes ?? new TreeNodeCollection();
```

Comment:
Because of this return, we will skip the last call to `GetTreeNodesInternal` which will get the list of - non root -  documents that the user has access to.

And as far as I have seen, it means that the user only sees the Recycle Bin and not their actual content nodes anymore.

![image](https://github.com/umbraco/Umbraco-CMS/assets/1769475/0463602b-673a-4bdd-87c7-bdc0571fdc4a)

So, my impression is that we should probably get the internal nodes, and then add the recycle bin to the collection if the condition is met, not have one or the other.

What do you think? Does this all make sense, or did I maybe oversee something ? 😅

Reply:
Thanks for the feedback.  You were spot on.  I'd messed up the permissions on my test group, so had missed this during my previous testing.

I've rewritten the code so everyone will now get access to the nodes they've been granted permissions to, as well as the recycle bin even if they don't have permission to access the root content node.